gme/Ay_Cpu.h

```diff
@@ -4,7 +4,7 @@
 #ifndef AY_CPU_H
 #define AY_CPU_H
 
-#include "blargg_endian.h"
```

Comment:
Patch is big. But at least this particular bit is wrong: This header references endian macros, but it now doen't include blargg_endian.h, and in Ay_Cpu.c where it is included blargg_endian.h is included after Ay_Cpu.h

There may be other breakages, too...

Reply:
I'm not sure I follow. The header doesn't use `blargg_endian.h` but the source file does.

What happens as a result of `blargg_endian.h` being included after `Ay_Cpu.h`?

Reply:
Build with -Wundef to see: In short, big endian define will be missing and builds on big endian arches will be wrong

Reply:
When you say big endian define, are you referring to `BLARGG_BIG_ENDIAN`?

If so, that's not defined in `blargg_endian.h`, that appears to be defined by the cmake script, or something equivalent.

Reply:
> When you say big endian define, are you referring to `BLARGG_BIG_ENDIAN`?

Yes

> If so, that's not defined in `blargg_endian.h`, that gets defined by the build system.

Well, you're actually correct (and I was the one who did it that way.. face palm..)

> It looks like a lot of these undef warnings could be remedied by simply changing `#if` to `#ifdef/#ifndef`

Be _really_ careful there.

And I hope that you don't include such a changeset in here: One patch, doing only one thing at a time should be the norm.